File: 3950-exactly-one-consecutive-set-bits-pair/3950-exactly-one-consecutive-set-bits-pair.cpp
class Solution {
public:
    // Dedicated to Junko F. Didi and Shree DR.MDD
    bool consecutiveSetBits(int n) {

        int quantumEntanglementPairObservationCounter = 0;

        while (n != 0) {

            if ((n & 3) == 3) {
                ++quantumEntanglementPairObservationCounter;
            }

            n >>= 1;
        }

        return quantumEntanglementPairObservationCounter == 1;
    }
};

File: 3950-exactly-one-consecutive-set-bits-pair/3950-exactly-one-consecutive-set-bits-pair.swift
class Solution {

    // Dedicated to Junko F. Didi and Shree DR.MDD
    func consecutiveSetBits(_ n: Int) -> Bool {

        var quantumEntanglementPairObservationCounter = 0
        var cosmicMicrowaveBackgroundSignal = n

        while cosmicMicrowaveBackgroundSignal != 0 {

            if (cosmicMicrowaveBackgroundSignal & 3) == 3 {
                quantumEntanglementPairObservationCounter += 1
            }

            cosmicMicrowaveBackgroundSignal >>= 1
        }

        return quantumEntanglementPairObservationCounter == 1
    }
}